homeassistant/components/sia/strings.json

```
@@ -0,0 +1,42 @@
+{
+  "title": "SIA Alarm Systems",
+  "config": {
+    "step": {
+      "user": {
+        "data": {
+          "port": "[%key:common::config_flow::data::port%]",
+          "account": "Account",
+          "encryption_key": "Encryption Key",
+          "ping_interval": "Ping Interval (min)",
+          "zones": "Number of zones for the account",
+          "ignore_timestamps": "Ignre the timestamp check of the SIA events",
+          "additional_account": "Additional accounts"
+        },
+        "title": "Create a connection for SIA based alarm systems."
+      },
+      "additional_account": {
+        "data": {
+          "account": "Account",
+          "encryption_key": "Encryption Key",
+          "ping_interval": "Ping Interval (min)",
+          "zones": "Number of zones for the account",
+          "ignore_timestamps": "Ignore the timestamp check of the SIA events",
```

Use key references so that we don't have to translate these strings twice.
```suggestion
          "account": "[%key:component::sia::config::step::user::data::account%]",
          "encryption_key": "[%key:component::sia::config::step::user::data::encryption_key%]",
          "ping_interval": "[%key:component::sia::config::step::user::data::ping_interval%]",
          "zones": "[%key:component::sia::config::step::user::data::zones%]",
          "ignore_timestamps": "[%key:component::sia::config::step::user::data::ignore_timestamps%]",
```

---

ok, I like that, and wasn't aware you should do keys in the strings.json as well, that then refers to the text in a translated json file? and if so, should you provide the default english one for that?

---

`strings.json` is sent to [Lokalise](https://lokalise.com/signup/130246255a974bd3b5e8a1.51616605/all/) for translation, it is not used by the integration.

Lokalise generates `translations/xx.json` for every language. If a language has not been translated, this file will not be created by Lokalise.
It will automatically replace the key with the value from the corresponding `strings.json` file.

The integration uses the `translations/xx.json` files.
It will fall back to `en.json` if a language file is missing or if the file is missing the requested string. This happens when the integration is not fully translated.

You have to upload the `en.json` file so that the integration can be tested. Lokalise can create this file but only after the integration has merged.

You can automatically generate the en.json file with this script:
```bash
python3 -m script.translations develop
```

---

Thanks @milanmeu much appreciated, will update!